pingcap/tidb
# ddl: drop placement rule cache when droping database

### What problem does this PR solve?

close #22364

Problem Summary: `drop database xx` statements won't drop placement rules accordingly. 

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

```
mycli -h 127.0.0.1 -P 4000 -u root --no-warn << "EOF"
set global tidb_enable_alter_placement=on;
drop database if exists test;
create database test;
EOF

mycli -h 127.0.0.1 -P 4000 -u root --no-warn << "EOF"
set global tidb_enable_alter_placement=on;
create table test.t1 (a int) partition by hash(a) partitions 2;
alter table test.t1 alter partition p0 alter placement policy role=leader;
EOF

pd-ctl config placement-rules show // the leader rule should show

mycli -h 127.0.0.1 -P 4000 -u root --no-warn << "EOF"
drop database test;
EOF

pd-ctl config placement-rules show // only the default rule
```

### Release note 

- drop database statements will drop placement rules of the database, its tables, and partitions of tables.
